Split out ios shared feed protos

chrome_feed_response_metadata.proto is needed in ios so split it out
so it can be separately compiled into ios.

Change-Id: I9e5e50f5e1742258f5c8197d54f0c610b56daf4b
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4266050
Commit-Queue: edchin <edchin@google.com>
Reviewed-by: Dan H <harringtond@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1107357}
diff --git a/components/feed/core/proto/BUILD.gn b/components/feed/core/proto/BUILD.gn
index 51916e08..f0ec531 100644
--- a/components/feed/core/proto/BUILD.gn
+++ b/components/feed/core/proto/BUILD.gn
@@ -22,7 +22,6 @@
     "v2/wire/capabilities_debug_data.proto",
     "v2/wire/capability.proto",
     "v2/wire/chrome_client_info.proto",
-    "v2/wire/chrome_feed_response_metadata.proto",
     "v2/wire/chrome_fulfillment_info.proto",
     "v2/wire/client_info.proto",
     "v2/wire/client_user_profiles.proto",
@@ -69,6 +68,10 @@
   ]
 }
 
+proto_library("proto_ios_shared_v2") {
+  sources = [ "v2/wire/chrome_feed_response_metadata.proto" ]
+}
+
 if (is_android) {
   proto_java_library("proto_java_v2") {
     proto_path = "../../../../"
diff --git a/components/feed/core/v2/BUILD.gn b/components/feed/core/v2/BUILD.gn
index ab5efcc4..b1d3fb4 100644
--- a/components/feed/core/v2/BUILD.gn
+++ b/components/feed/core/v2/BUILD.gn
@@ -172,6 +172,7 @@
     ":ios_shared",
     "//base",
     "//components/feed/core/common:feed_core_common",
+    "//components/feed/core/proto:proto_ios_shared_v2",
     "//components/feed/core/proto:proto_v2",
   ]
 }
@@ -187,7 +188,7 @@
   deps = [
     "//base",
     "//components/feed/core/common:feed_core_common",
-    "//components/feed/core/proto:proto_v2",
+    "//components/feed/core/proto:proto_ios_shared_v2",
     "//components/prefs",
   ]
 }
diff --git a/components/feed/core/v2/public/ios/BUILD.gn b/components/feed/core/v2/public/ios/BUILD.gn
index 106f089a..59bdd2b 100644
--- a/components/feed/core/v2/public/ios/BUILD.gn
+++ b/components/feed/core/v2/public/ios/BUILD.gn
@@ -25,7 +25,6 @@
   deps = [
     ":feed_ios_public",
     "//base/test:test_support",
-    "//components/feed/core/v2:feed_core_v2",
     "//components/prefs",
     "//components/prefs:test_support",
     "//testing/gmock",
